### PR TITLE
feat: Adding well-known URI

### PR DESCRIPTION
**Pull request by nguyen102**
Issue #, if available:

Description of changes:
Adding `.well-known/smart-configuration` [route](http://www.hl7.org/fhir/smart-app-launch/conformance/index.html#using-well-known)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/router/handlers/wellKnownUriHandler.test.ts [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import { SmartStrategy } from 'fhir-works-on-aws-interface';
+import { camelToSnakeCase, getWellKnownUriResponse } from './wellKnownUriHandler';
+
+describe('camelToSnakeCase', () => {
+    test('one word', () => {
+        expect(camelToSnakeCase('red')).toEqual('red');
+    });
```
Comment (rsmayda) [inline src/router/handlers/wellKnownUriHandler.test.ts:7]: Add a mis-capitalized test case
```suggestion
    test('one word', () => {
        expect(camelToSnakeCase('red')).toEqual('red');
    });
     test('one word; mis-capitalized ', () => {
        expect(camelToSnakeCase('Red')).toEqual('red');
    });
```

File: src/router/metadata/cap.rest.security.template.ts [hunk 2]
```diff
@@ -30,11 +30,11 @@ export default function makeSecurity(authConfig: Auth, hasCORSEnabled: boolean =
                             extension: [
                                 {
                                     url: 'token',
```
Comment (rsmayda) [inline src/router/metadata/cap.rest.security.template.ts]: This is out of scope but could you add a `// TODO` (or do it) to add the other endpoints to the cap statement? http://hl7.org/fhir/smart-app-launch/conformance/index.html#declaring-support-for-oauth2-endpoints

**Review by rsmayda — CHANGES_REQUESTED** [review]
Add a minor test

File: src/router/handlers/wellKnownUriHandler.test.ts (hunk 1 above)
Comment (nguyen102) [reply src/router/handlers/wellKnownUriHandler.test.ts:7]: Added this test case, and fixed the code to handle this edge case.

File: src/router/metadata/cap.rest.security.template.ts [hunk 3]
```diff
@@ -20,23 +20,50 @@ export default function makeSecurity(authConfig: Auth, hasCORSEnabled: boolean =
                 },
             ],
         };
-        if (authConfig.strategy.oauth) {
+        const { strategy } = authConfig;
+        const { oauthPolicy } = strategy;
```
Comment (rsmayda) [inline src/router/metadata/cap.rest.security.template.ts]: I don't see 'strategy' used anywhere?
```suggestion
        const { oauthPolicy } = authConfig.strategy;
```

File: src/router/metadata/cap.rest.security.template.ts [hunk 4]
```diff
@@ -20,23 +20,50 @@ export default function makeSecurity(authConfig: Auth, hasCORSEnabled: boolean =
...
+                    url: 'revoke',
+                    valueUri: oauthPolicy.revocationEndpoint,
+                });
+            }
+            if (oauthPolicy.registrationEndpoint) {
+                extension.push({
+                    url: 'register',
+                    valueUri: oauthPolicy.registrationEndpoint,
+                });
+            }
```
Comment (rsmayda) [inline src/router/metadata/cap.rest.security.template.ts:58]: Add at least 1 new test case for a scenario where they are all provided

File: src/router/metadata/cap.rest.security.template.ts (hunk 3 above)
Comment (nguyen102) [reply src/router/metadata/cap.rest.security.template.ts]: updated

File: src/router/metadata/cap.rest.security.template.ts (hunk 4 above)
Comment (nguyen102) [reply src/router/metadata/cap.rest.security.template.ts:58]: updated

**Review by rsmayda — APPROVED** [review]
